### ISSUE TITLE
gradient_override_map for tf.distributions.Bernoulli

### ISSUE THREAD

**Issue by joshivm** [tensorflow/tensorflow#18673]
I want to override gradient calculation for a series of ops that computes

`v = tf.reduce_mean(tf.distributions.Bernoulli(prob=[x]*10).sample())`

Here, shape of v is same as shape of x and I want dv/dx = input gradient at v

Now I know one can register gradients for a single op but how about a series of ops. If I try to register gradient for individual ops in the series I get error with shapes of input and gradient at one of the ops in the series.

Update 1:
I have written custom code for this purpose
OS Platform and Distribution I am using is Ubuntu 16.04 LTS
TensorFlow installed from - pip 
TensorFlow version - 1.5.0
Bazel version - NA
CUDA/cuDNN version - CuDNN9
GPU model and memory - GForce GTX 1060TI
Exact command to reproduce - NA

Is this possible in tensorflow?
Thanks in advance!!

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by joshivm** [comment]
Update 1:
I have written custom code for this purpose
OS Platform and Distribution I am using is Ubuntu 16.04 LTS
TensorFlow installed from - pip
TensorFlow version - 1.5.0
Bazel version - NA
CUDA/cuDNN version - CuDNN9
GPU model and memory - GForce GTX 1060TI
Exact command to reproduce - NA

**Comment by angersson** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!
